builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,211 @@
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+  - pipeline: images
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.1'
+    trigger:
+      branches:
+      - release/1.1
+  - pipeline: packages
+    source: 'Azure-IoT-Edge-Core Edgelet Packages'
+    branch: 'release/1.1'
+    trigger:
+      branches:
+      - release/1.1
+  - pipeline: images_main
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'main'
+  - pipeline: images_release_1_2
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.2'    
+
+jobs:
+################################################################################
+  - job: partial_upgrade_ubuntu_1804_msmoby
+################################################################################
+    displayName: Partial Upgrade
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
+
+    variables:
+      os: linux
+      arch: amd64
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+      minimal: true
+
+    strategy:
+      matrix:
+        1.1_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "release/1.1"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.1_to_1.3_no_runtime_modules:
+          edgeDaemonVersion: "release/1.1"
+          bootstrapAgentVersion: "release/1.1"
+          runtimeModulesVersion: "main"
+        1.1_to_1.2_no_edge_daemon:
+          edgeDaemonVersion: "release/1.1"
+          bootstrapAgentVersion: "release/1.2"
+          runtimeModulesVersion: "release/1.2"
+        1.1_to_1.2_only_runtime_modules:
+          edgeDaemonVersion: "release/1.1"
+          bootstrapAgentVersion: "release/1.1"
+          runtimeModulesVersion: "release/1.2"
+        1.1.0_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.1.0"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.1.0_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.1.0"
+          bootstrapAgentVersion: "1.1.0"
+          runtimeModulesVersion: "main"
+        1.1.8_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.1.8"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.1.8_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.1.8"
+          bootstrapAgentVersion: "1.1.8"
+          runtimeModulesVersion: "main"
+        1.1.13_to_1.3_no_edge_daemon:
+          edgeDaemonVersion: "1.1.13"
+          bootstrapAgentVersion: "main"
+          runtimeModulesVersion: "main"
+        1.1.13_to_1.3_only_runtime_modules:
+          edgeDaemonVersion: "1.1.13"
+          bootstrapAgentVersion: "1.1.13"
+          runtimeModulesVersion: "main"
```

Comment:
There's effectively no difference between `1.1.13_to_1.3_*` and `1.1_to_1.3_*` strategies, right? And when we add more 1.1.x releases we would need to update these strategies. Unless I'm missing something, I think we can simplify here by keeping the `1.1_to_1.3_*` strategies and removing the `1.1.13_to_1.3_*` strategies.

Reply:
Oh, I see the difference now...`1.1_to_1.3_*` strategies pull daily builds of edgelet from DevOps artifacts, whereas `1.1.13_to_1.3_*` strategies pull released versions of edgelet from packages.microsoft.com. This is fine.